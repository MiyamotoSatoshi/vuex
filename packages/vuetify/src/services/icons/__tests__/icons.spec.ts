// Service
import { Icons } from '../index'

describe('Icons.ts', () => {
  let icon

  it('should generate default icons', () => {
    icon = new Icons()

    expect(icon.values).toMatchSnapshot()
  })

  it('should use a custom iconfont preset', () => {
    icon = new Icons({
      iconfont: 'fa4',
    })

    expect(icon.values).toMatchSnapshot()
  })

  it('should accept custom icons', () => {
    icon = new Icons({
      values: {
        complete: 'fizzbuzz',
      },
    })

    expect(icon.values.complete).toBe('fizzbuzz')
  })
})
