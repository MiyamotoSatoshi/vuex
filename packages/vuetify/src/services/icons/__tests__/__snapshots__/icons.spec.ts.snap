// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icons.ts should generate default icons 1`] = `
Object {
  "cancel": "cancel",
  "checkboxIndeterminate": "indeterminate_check_box",
  "checkboxOff": "check_box_outline_blank",
  "checkboxOn": "check_box",
  "clear": "clear",
  "close": "close",
  "complete": "check",
  "delete": "cancel",
  "delimiter": "fiber_manual_record",
  "dropdown": "arrow_drop_down",
  "edit": "edit",
  "error": "warning",
  "expand": "keyboard_arrow_down",
  "info": "info",
  "loading": "cached",
  "menu": "menu",
  "next": "chevron_right",
  "prev": "chevron_left",
  "radioOff": "radio_button_unchecked",
  "radioOn": "radio_button_checked",
  "ratingEmpty": "star_border",
  "ratingFull": "star",
  "ratingHalf": "star_half",
  "sort": "arrow_upward",
  "subgroup": "arrow_drop_down",
  "success": "check_circle",
  "warning": "priority_high",
}
`;

exports[`Icons.ts should use a custom iconfont preset 1`] = `
Object {
  "cancel": "fa fa-times-circle",
  "checkboxIndeterminate": "fa fa-minus-square",
  "checkboxOff": "fa fa-square-o",
  "checkboxOn": "fa fa-check-square",
  "clear": "fa fa-times-circle",
  "close": "fa fa-times",
  "complete": "fa fa-check",
  "delete": "fa fa-times-circle",
  "delimiter": "fa fa-circle",
  "dropdown": "fa fa-caret-down",
  "edit": "fa fa-pencil",
  "error": "fa fa-exclamation-triangle",
  "expand": "fa fa-chevron-down",
  "info": "fa fa-info-circle",
  "menu": "fa fa-bars",
  "next": "fa fa-chevron-right",
  "prev": "fa fa-chevron-left",
  "radioOff": "fa fa-circle-o",
  "radioOn": "fa fa-dot-circle",
  "ratingEmpty": "fa fa-star-o",
  "ratingFull": "fa fa-star",
  "ratingHalf": "fa fa-star-half-o",
  "sort": "fa fa-sort-up",
  "subgroup": "fa fa-caret-down",
  "success": "fa fa-check-circle",
  "warning": "fa fa-exclamation",
}
`;
